### PR TITLE
[MAINT] refactor nilearn/surface (sourcery, f-strings)

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to https://github.com/nilearn/nilearn/issues/2528

Changes proposed in this pull request:

- apply sourcery refactoring suggestions
- use f strings everywhere

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM.

File: nilearn/surface/surface.py [hunk 1]
```diff
@@ -727,12 +746,14 @@ def load_surf_data(surf_data):
...
-                                  'or one of the following file formats: .gii,'
-                                  ' .gii.gz, .mgz, .nii, .nii.gz, Freesurfer '
-                                  'specific files such as .area, .curv, .sulc,'
-                                  ' .thickness, .annot, .label') % surf_data)
+                raise ValueError(
+                    'The input type is not recognized. '
+                    f'{surf_data!r} was '
+                    'given while valid inputs are a Numpy array '
+                    'or one of the following file formats: .gii,'
+                    ' .gii.gz, .mgz, .nii, .nii.gz, '
```
Comment (ymzayek) [inline nilearn/surface/surface.py:754]: Take it or leave it: This list of extensions also appears 4 times so why not standardize is like the others

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @Remi-Gau !

File: nilearn/surface/surface.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/surface/surface.py:754]: yeah I was very tempted but felt like I was getting on a slippery slope of "but then I could also refactor this, and also this, and that, and this there..."

will do this one though. :-p

Comment (Remi-Gau) [reply nilearn/surface/surface.py]: Done. But of course I had do also hoist a couple of guard clauses because it was just too tempting.

**Pull request merged**